include/libdnf-cli/argument_parser.hpp

```diff
@@ -30,61 +30,135 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+class ArgumentParserGroupArgumentIdRegisteredError : public ArgumentParserError {
+public:
+    using ArgumentParserError::ArgumentParserError;
+    const char * get_name() const noexcept override { return "ArgumentParserGroupArgumentIdRegisteredError"; }
+};
+
+
+/// Exception is thrown when the Argument `id` contains not allowed '.' character.
+class ArgumentParserArgumentInvalidIdError : public ArgumentParserError {
+public:
+    using ArgumentParserError::ArgumentParserError;
+    const char * get_name() const noexcept override { return "ArgumentParserArgumentInvalidIdError"; }
+};
+
+
+/// Exception is thrown when there are insufficient values for the positional argument.
+class ArgumentParserPositionalArgFewValuesError : public ArgumentParserError {
+public:
+    using ArgumentParserError::ArgumentParserError;
+    const char * get_name() const noexcept override { return "ArgumentParserPositionalArgFewValuesError"; }
+};
+
+
+/// Exception is thrown if the named argument requires a value and the value is missing.
+class ArgumentParserNamedArgMissingValueError : public ArgumentParserError {
+public:
+    using ArgumentParserError::ArgumentParserError;
+    const char * get_name() const noexcept override { return "ArgumentParserNamedArgMissingValueError"; }
+};
+
+/// Exception is thrown if the named argument is defined without a value and a value is present.
+class ArgumentParserNamedArgUnexpectedValueError : public ArgumentParserError {
+public:
+    using ArgumentParserError::ArgumentParserError;
+    const char * get_name() const noexcept override { return "ArgumentParserNamedArgUnexpectedValueError"; }
+};
+
+
+/// Exception is thrown when a command was not found.
+class ArgumentParserCommandNotFoundError : public ArgumentParserError {
+public:
+    using ArgumentParserError::ArgumentParserError;
+    const char * get_name() const noexcept override { return "ArgumentParserCommandNotFoundError"; }
+};
+
+/// Exception is thrown when a named argument was not found.
+class ArgumentParserNamedArgNotFoundError : public ArgumentParserError {
+public:
+    using ArgumentParserError::ArgumentParserError;
+    const char * get_name() const noexcept override { return "ArgumentParserNamedArgNotFoundError"; }
+};
+
+/// Exception is thrown when a positional argument was not found.
+class ArgumentParserPositionalArgNotFoundError : public ArgumentParserError {
+public:
+    using ArgumentParserError::ArgumentParserError;
+    const char * get_name() const noexcept override { return "ArgumentParserPositionalArgNotFoundError"; }
+};
+
+/// Exception is thrown when a command with the same ID is already registered.
+class ArgumentParserCommandCommandIdRegisteredError : public ArgumentParserError {
+public:
+    using ArgumentParserError::ArgumentParserError;
+    const char * get_name() const noexcept override { return "ArgumentParserCommandCommandIdRegisteredError"; }
+};
```

Comment:
Programmer error? I was thinking about it. User installs/loads two microdnf plugins that implement a command with the same `id`. Is it a programmer error or user error?

Reply:
Ok, if it would come from the plugins that would be in runtime, a user error... these are hard to judge.

Reply:
@jrohel hmm, the comment as well as the 3 similar comments below actually also suggest to rename the exception class.

And (once again), could we actually just have one exception class for all the "already registered" cases called `ArgumentParserAlreadyRegistered`? The name would actually be sane, there will be less fragmentation and the exceptions are thrown from different methods so I think it would be a strictly better API.

Same reduction could be applied to `ArgumentParserCommandNotFoundError` and perhaps others. I'd like to reduce the amount of useless code...

Reply:
I renamed the exception classes and reduced the number of classes.
Three clases were replaced by one `ArgumentParserNotFoundError`.
Four classes were replaced by one `ArgumentParserIdAlreadyRegisteredError`.